src/hooks/useVote.js

```diff
@@ -0,0 +1,192 @@
+import { useEffect, useState, useMemo } from 'react'
+import { connectorConfig } from '../current-environment'
+import { createAppHook } from '@aragon/connect-react'
+import { formatTokenAmount } from '@aragon/ui'
+import connectVoting from '@aragon/connect-disputable-voting'
+import { useWallet } from '../providers/Wallet'
+import { useMounted } from '../hooks/useMounted'
+import { ProposalNotFound } from '../errors'
+import { useOrgApps } from '../providers/OrgApps'
+
+const useDisputableVoting = createAppHook(
+  connectVoting,
+  connectorConfig.disputableVoting
+)
+
+export function useVote(proposalId) {
+  const [vote, loading] = useVoteSubscription(proposalId)
+  return [vote, loading]
+}
+
+function useVoteSubscription(proposalId) {
+  const { account } = useWallet()
+  const { disputableVotingApp } = useOrgApps()
+
+  const [
+    vote,
+    { loading: voteLoading, error: voteError },
+  ] = useDisputableVoting(
+    disputableVotingApp,
+    (app) => app.onVote(`${disputableVotingApp.address}-vote-${proposalId}`),
+    [proposalId]
+  )
+
+  const voteUpdateValue = JSON.stringify(vote)
+
+  // This is a workaround for receiving the latest vote after an update.
+  // Currently just listening for a vote change isn't enough to get all of the latest changes
+  // so we use a subscription to castVote which correctly triggers on a full update
+  // and use the return value as a dependency when passing down the vote.
+  const [
+    castVote,
+    { error: castVoteError },
+  ] = useDisputableVoting(
+    disputableVotingApp,
+    () => (vote && account ? vote.onCastVote(account) : null),
+    [account, voteUpdateValue]
+  )
+
+  // Use caseVote as a dependency value for storing the latest vote
+  const castVoteUpdateValue = JSON.stringify(castVote)
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  const latestVote = useMemo(() => vote, [castVoteUpdateValue])
+  /* eslint-enable react-hooks/exhaustive-deps */
```

Comment:
This is pretty awful in general, it basically uses the `onCastVote` subscription to decide when to refresh the `vote` object and trigger dependencies.

Ideally we would want to be subscribing to each value individually and only perform the required work based on what has changed, this isn't really possible to do at the moment though.